packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js

```diff
@@ -1,15 +1,63 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import locationValidator from '../../helpers/locationValidator'
+import asyncStates from 'src/helpers/asyncStates'
 
-function SingleImageViewer ({ subject }) {
-  const url = Object.values(subject.locations[0])[0]
+class SingleImageViewer extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      height: null,
+      width: null,
+      loading: asyncStates.initialized
+    }
+  }
 
-  return (
-    <svg>
-      <image xlinkHref={url} />
-    </svg>
-  )
+  componentDidMount () {
+    this.handleSubject()
+  }
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.subject.id !== this.props.subject.id) {
```

Comment:
This should have a default prop for `subject` and/or do an existential check for `prevProps.subject` and `this.props.subject`